controllers/workspace/provision/routing.go

```diff
@@ -144,15 +145,14 @@ func getSpecRouting(
 
 	var annotations map[string]string
 	if val, ok := workspace.Annotations[config.WorkspaceRestrictedAccessAnnotation]; ok {
-		annotations = map[string]string{}
-		annotations[config.WorkspaceRestrictedAccessAnnotation] = val
+		maputils.Append(annotations, config.WorkspaceRestrictedAccessAnnotation, val)
```

Comment:
```suggestion
		annotations = maputils.Append(annotations, config.WorkspaceRestrictedAccessAnnotation, val)
```
you're caught on the confusion you predicted )

Reply:
Squashed into b40bf35eb061c12ae61b1825b072ef2778a44886